var/spack/repos/builtin/packages/py-python-lsp-jsonrpc/package.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyPythonLspJsonrpc(PythonPackage):
+    """A Python 3.7+ server implementation of the JSON RPC 2.0 protocol."""
+
+    homepage = "https://github.com/python-lsp/python-lsp-jsonrpc"
+    pypi = "python-lsp-jsonrpc/python-lsp-jsonrpc-1.0.0.tar.gz"
+
+    maintainers = ["alecbcs"]
+
+    version("1.0.0", sha256="7bec170733db628d3506ea3a5288ff76aa33c70215ed223abdb0d95e957660bd")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools@61.2.0:", type="build")
```

Comment:
Where do these version constraints come from?

Reply:
Haven't packaged too many python applications before outside of pypi, so I was looking at the project's [pyproject.toml](https://github.com/python-lsp/python-lsp-server/blob/develop/pyproject.toml) file. That contains the version constraints under the build-system.requires property. Is this not the right place/are those constraints created as part of a template?

Reply:
Oh and the general` python@3.7:` I got from the description of the project.
> A Python 3.7+ implementation of the [Language Server Protocol](https://github.com/Microsoft/language-server-protocol).

Reply:
That file is present on develop but not in 1.0.0, that's why I'm confused.

Reply:
Ah! Wow you're right I'd assumed since the project seems fairly stable/inactive that the pyproject.toml would have been there at the 1.0.0 release but you're right that's not the case! I'll relax the constraints and see if it continues to build correctly.